client/x/evmstaking/keeper/abci.go

```diff
@@ -78,11 +78,6 @@ func (k *Keeper) EndBlock(ctx context.Context) (abci.ValidatorUpdates, error) {
 		}
 	}
 
-	valUpdates, err := k.stakingKeeper.EndBlocker(ctx)
```

Comment:
I think we cannot move valUpdates logic to the end. It has to be before adding the unbonded entries to withdraw queue. 

The reason is that: 
During staking keeper's EndBlocker, it unbonds the mature unbending delegations and release the delegated token. Only after the token release, we can do the next step, which is to burn these delegated token and prepare to mint them to EL side.

Reply:
Ah, I see. you're right. I will move it before handling unbondedEntries.